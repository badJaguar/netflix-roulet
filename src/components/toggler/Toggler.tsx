import React from "react";

import { ErrorBoundry } from "../../services/ErrorBoundry";
import { ITogglerProps } from "./ITogglerProps";

export class Toggler extends React.Component<ITogglerProps> {

    public render() {

        const { idLeft, idRight, leftLabel, name, onChange, rightLabel, title, toggle } = this.props;

        return (
            <ErrorBoundry>
                <div className="searchFilter">
                    {title} &nbsp; &nbsp;
                    <div className="switchField">
                        <input
                            type="radio"
                            id={idLeft}
                            name={name}
                            value={leftLabel}
                            onChange={onChange}
                            checked={!toggle}
                        />
                        <label htmlFor={idLeft}>{leftLabel}</label>

                        <input
                            type="radio"
                            id={idRight}
                            name={name}
                            value={rightLabel}
                            onChange={onChange}
                            checked={toggle}
                        />
                        <label htmlFor={idRight}>{rightLabel}</label>
                    </div>
                </div>
            </ErrorBoundry >
        );
    }
}
