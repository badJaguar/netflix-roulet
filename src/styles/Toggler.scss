@import './colors.scss';

.searchFilter {
    color: $white;
    display: inline-flex;
    flex-direction: row;
    width: 100%;
    align-items: baseline;
    margin-top: 2vh;
}

.filterButtonGroup {
    width: 4vh;
    display: flex;

}

.switchField input {
    position: absolute !important;
    clip: rect(0, 0, 0, 0);
    height: 1px;
    width: 1px;
    border: 0;
    overflow: hidden;
}

.switchField label {
    background-color: #e4e4e4;
    color: rgba(0, 0, 0, 0.6);
    font-size: 14px;
    line-height: 1;
    text-align: center;
    padding: 8px 16px;
    margin-right: -1px;
    border: 1px solid rgba(0, 0, 0, 0.2);
}

.switchField label:hover {
    cursor: pointer;
}

.switchField input:checked+label {
    background-color: $red;
    box-shadow: none;
}

.switchField label:first-of-type {
    border-radius: 4px 0 0 4px;
    background-color: rgba(66, 66, 66, 0.68);
    color: $white;
}

.switchField label:last-of-type {
    border-radius: 0 4px 4px 0;
    background-color: rgba(66, 66, 66, 0.68);
    color: $white;

}