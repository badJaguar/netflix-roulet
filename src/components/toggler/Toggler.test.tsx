import { shallow } from "enzyme";
import * as React from "react";
import { Toggler } from "./Toggler";

const onClick = jest.fn();
test("Toggler changes the text after click", () => {
    const toggler = shallow(<Toggler
        leftLabel="TITLE"
        rightLabel="GENRE"
        title={""}
        toggle={false}
        idLeft={""}
        idRight={""}
        name={""}
        onChange={() => onClick}
    />);

    toggler.find("value").contains("this.props.leftLabel");
    toggler.findWhere((input) => input.contains("this.props.idLeft"));

    expect(toggler).toMatchSnapshot();
});
