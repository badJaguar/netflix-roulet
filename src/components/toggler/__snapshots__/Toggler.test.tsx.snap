// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggler changes the text after click 1`] = `
<ErrorBoundry>
  <div
    className="searchFilter"
  >
        
    <div
      className="switchField"
    >
      <input
        checked={true}
        id=""
        name=""
        onChange={[Function]}
        type="radio"
        value="TITLE"
      />
      <label
        htmlFor=""
      >
        TITLE
      </label>
      <input
        checked={false}
        id=""
        name=""
        onChange={[Function]}
        type="radio"
        value="GENRE"
      />
      <label
        htmlFor=""
      >
        GENRE
      </label>
    </div>
  </div>
</ErrorBoundry>
`;
